api/envoy/config/listener/v3/udp_listener_config.proto

```diff
@@ -35,11 +36,8 @@ message UdpListenerConfig {
   // [#comment:TODO(#12829): Remove this as an extension point.]
   core.v3.TypedExtensionConfig listener_config = 4;
 
-  // The maximum size of received downstream UDP datagrams. Using a larger size will cause Envoy to allocate
-  // more memory per listener. Received datagrams above this size will be dropped. If not set
-  // defaults to 1500 bytes.
-  google.protobuf.UInt64Value max_downstream_rx_datagram_size = 5
-      [(validate.rules).uint64 = {lt: 65536 gt: 0}];
+  // UDP socket configuration for the listener.
+  core.v3.UdpSocketConfig downstream_socket_config = 5;
```

Comment:
Will everything that ever goes into this message have a sane default value? If not, maybe it should be required? I'm not sure. @envoyproxy/api-shepherds may have thoughts.

Reply:
My thinking is everything will have a default, though it's possible the default will be context specific in the future, so IMO it's fine to leave this without being required but will wait to hear from others.

Reply:
Actually, I think anything must have a default value, because you can't add a new required field to an old message without breaking compatibility. So I think this is fine as-is.

Reply:
Yeah that's true.